crates/bdk/src/wallet/coin_selection.rs

```diff
@@ -120,8 +120,9 @@ use rand::seq::SliceRandom;
 pub type DefaultCoinSelectionAlgorithm = BranchAndBoundCoinSelection;
 
 // Base weight of a Txin, not counting the weight needed for satisfying it.
-// prev_txid (32 bytes) + prev_vout (4 bytes) + sequence (4 bytes)
-pub(crate) const TXIN_BASE_WEIGHT: usize = (32 + 4 + 4) * 4;
+// prev_txid (32 bytes) + prev_vout (4 bytes) + sequence (4 bytes) + script_len
+// (1 byte).
+pub(crate) const TXIN_BASE_WEIGHT: usize = (32 + 4 + 4 + 1) * 4;
```

Comment:
@evanlinjin You mean something like this?

Reply:
You're missing the witness length